### ISSUE TITLE
[Feature Request]: Use the WinForms Designer Extensibility SDK for .NET

### ISSUE THREAD

**Issue by PWagner1** [Krypton-Suite/Standard-Toolkit#593]
Since MS is having issues supporting the `WinForms` designer in .NET, the WinForms Designer Extensibility SDK could be the solution for the current [designer issues](https://visualstudiomagazine.com/articles/2022/01/13/winforms-designer-parity.aspx)

**Comment by Smurf-IV** [comment]
Nice find, All Designer issues should be linked to this.

**Comment by Wagnerp** [comment]
> Nice find, All Designer issues should be linked to this.

Well it's actually @sdreb3421 who notified me about this.

**Comment by Smurf-IV** [comment]
Also, (Maybe - it's unclear) a reason some of the .Net5 is not working in a designer might be down to the loading in .Net5Core which has quite a few of the following "Color API's" missing:
https://github.com/dotnet/runtime/issues/18476

**Comment by Smurf-IV** [comment]
@Wagnerp This is going to be "Interesting"
Quote: _The designers, as per the API specs, need to be implemented inside a NuGet package._

See the middle of the following page: 
https://www.telerik.com/blogs/new-telerik-ui-for-winforms-designers-dotnet-core-dotnet-5

**Comment by Smurf-IV** [comment]
And the following Caveats and "Tips'n'tricks" will need to be reformatted to reflect whatever "we" get working for Krypton:
https://docs.telerik.com/devtools/winforms/core/design-time

**Comment by Wagnerp** [comment]
It's not ideal, but maybe private nightly packages could be used for testing?

**Comment by Smurf-IV** [comment]
I spent all day yesterday trying to find documentation on how to use this, and came up empty.
Telerik and devexpress mention it in their latest "suites", but that's it. :-(

**Comment by Wagnerp** [comment]
> I spent all day yesterday trying to find documentation on how to use this, and came up empty. Telerik and devexpress mention it in their latest "suites", but that's it. :-(

I think it's because it is so new, there is limited documentation so far. The packages only first appeared back in November.

**Comment by Smurf-IV** [comment]
I have had an inspiration from the "Read between the lines of why the above **is** solving things"

If I 
- Remove "ALL" designers from the toolkit
- Set toolkit project to be "Only" net5 windows
- Create a new winform net 5 windows only main app
- build.
- I get the following showing up in the designer:
![image](https://user-images.githubusercontent.com/2418812/149653283-14378710-e928-4c54-9e55-693cea4df479.png)

I'll need to investigate further to see "What" the _designers_ are attempting to do; but so far I've been able to drag (A random selection not all) controls onto the form !

**Comment by Wagnerp** [comment]
Just out of interest, do the action lists show up when using .NET?

**Comment by Smurf-IV** [comment]
> Just out of interest, do the action lists show up when using .NET?

Can you show a specific example, and do you mean in "Net 5" ?

**Comment by Wagnerp** [comment]
> > Just out of interest, do the action lists show up when using .NET?
> 
> Can you show a specific example, and do you mean in "Net 5" ?
> 
> The reason I ask is because of the following statement: ![image](https://user-images.githubusercontent.com/2418812/149653836-f3e595bd-1416-4cd1-bec1-a4d3a8f7407e.png) Even when AutoSize for a button is set to `true` it still has the Resize Handles ! i.e. there is no code that appears to be actually doing the statement in the designer of 4.8 !

Here's what I mean

![Animation](https://user-images.githubusercontent.com/949607/149654651-801296d0-973a-4424-887e-d563ab9b4790.gif)

**Comment by Smurf-IV** [comment]
Strange I never use those arrow thingies except for TableLayout
So if that is the only thing that is going to be missing (Because the properties will still show) then that can be "Lived" with for now ??
i.e. if not .48 or less than designer are disabled.
I can perform the above fix and relate it to the existing designer issues, and leave this one open until more documentation is available .
Thoughts ?

**Comment by Wagnerp** [comment]
> Strange I never use those arrow thingies except for TableLayout So if that is the only thing that is going to be missing (Because the properties will still show) then that can be "Lived" with for now ?? i.e. if not .48 or less than designer are disabled. I can perform the above fix and relate it to the existing designer issues, and leave this one open until more documentation is available . Thoughts ?

This might be a bug on MS part, maybe add this one specifically to the backlog until more documentation comes out?

**Comment by Smurf-IV** [comment]
> This might be a bug on MS part, 

Yes, that is what the mentioned (titled) sdk is trying to solve.

> maybe add this one specifically to the backlog until more documentation comes out?

Agreed.
This will lead to a set of related nuget packages, that will include just the designer for the project and will need to be set so that they do not become runtime references
Each TFM will need a dll to contain the designers for the suite element that is included.

**Comment by Smurf-IV** [comment]
Not sure when it appeared in the "Preview features" but this is in 2022.2.0 preview **3**:
![image](https://user-images.githubusercontent.com/2418812/163668460-0166d9a6-5963-466e-a00b-d65f76df1676.png)

**Comment by Smurf-IV** [comment]
https://www.nuget.org/packages/Microsoft.WinForms.Designer.SDK
"Still" in pre-release:
![image](https://user-images.githubusercontent.com/2418812/179359517-91a7830a-059a-4c28-9c84-9bd336fb7a6f.png)

**Comment by Smurf-IV** [comment]
Note this on the process of creating a designer to work in both:
https://www.codeproject.com/Articles/5341387/Writing-Custom-Control-in-NET-6

**Comment by Smurf-IV** [comment]
As seen in the latest 17.8.p3:
![image](https://github.com/Krypton-Suite/Standard-Toolkit/assets/2418812/16723fdf-22c7-4309-8655-cd24258fe8a0)

This is still not a "Feature" I want to implement until it is "Stable" in VS
Because the workaround - Just Works !

**Comment by Wagnerp** [comment]
Hi @Smurf-IV & @sneusse 

There has been movement in this area in VS 17.9

**Comment by Wagnerp** [comment]
@Smurf-IV I don't know if [this](https://devblogs.microsoft.com/visualstudio/winforms-designer-selection-for-32-bit-net-framework-projects/) is relevant?

**Comment by Camel-RD** [comment]
Maybe something like this could be usefull:
https://github.com/Camel-RD/KlonsLIB.Design

**Comment by mbagattini** [comment]
I'm struggling designing Krypton controls after upgrading a solution from .NET Framework 4.8 to .NET 6. Some controls simply refuse to be designed in the usual way (on the Windows Forms designer surface), I have to fiddle with the properties box and it's very frustrating. I checked the "preview" thing in the Environment section under controls but no luck. Is there any workaround to be able to design controls (for example, work with Krypton Ribbon) as it was under .NET 4.8?

**Comment by Wagnerp** [comment]
> I'm struggling designing Krypton controls after upgrading a solution from .NET Framework 4.8 to .NET 6. Some controls simply refuse to be designed in the usual way (on the Windows Forms designer surface), I have to fiddle with the properties box and it's very frustrating. I checked the "preview" thing in the Environment section under controls but no luck. Is there any workaround to be able to design controls (for example, work with Krypton Ribbon) as it was under .NET 4.8?

Hi @mbagattini 

There is a workaround, I've just updated the online documentation, but it won't be available until tomorrow when it'll be merged. The short of it is you need to set the `TargetFrameworks` to 'net48;netx.0-windows'. 'x' denotes the .NET version, 6, 8 or 9. Please do not use .NET 7, as support ends on 14th May 2024.

**Comment by Wagnerp** [comment]
Putting this [here](https://learn.microsoft.com/en-us/visualstudio/releases/2022/release-notes#windows-forms-out-of-process-designer) (VS 17.10)

**Comment by Smurf-IV** [comment]
MS are still having issues: This is still not fixed in 17.77.p3

**Comment by Smurf-IV** [comment]
@All Do "We" think that MS have stopped messing around with the external designer source usage for it to be stable enough to have a go at this now?
I have not seen any "Bug Fix / Change" announcements since 17.13.x (I have not been looking specifically, juts not noticed any coming through!)

**Comment by PWagner1** [comment]
> [@ALL](https://github.com/ALL) Do "We" think that MS have stopped messing around with the external designer source usage for it to be stable enough to have a go at this now? I have not seen any "Bug Fix / Change" announcements since 17.13.x (I have not been looking specifically, juts not noticed any coming through!)

@Smurf-IV I'd give it a go... I do plan to post a message on Discord soon for a project idea to bring the look & feel of Krypton to web devs, but need some thoughts...
